src/main/java/org/apache/sysml/runtime/instructions/cp/ConvolutionCPInstruction.java

```diff
@@ -297,6 +320,65 @@ public void processBiasMultiplyInstruction(ExecutionContext ec) throws DMLRuntim
 		ec.setMatrixOutput(getOutputVariableName(), outputBlock, getExtendedOpcode());
 	}
 	
+	public void processChannelSumsInstruction(ExecutionContext ec) throws DMLRuntimeException {
+		MatrixBlock input = ec.getMatrixInput(input1.getName(), getExtendedOpcode());
```

Comment:
Are you also going to add a multithreaded version for this later?

Reply:
No, it is purely a memory bound operation.